File: sobit_pro_library/test/test_arm_controll.py
#!/usr/bin/env python3
import rospy
from sobit_pro_module import SobitProJointController
from sobit_pro_module import Joint
from geometry_msgs.msg import Point
import sys

def test():
    rospy.init_node('test')
    r = rospy.Rate(1) # 10hz
    args = sys.argv
    pro_ctr = SobitProJointController(args[0]) # args[0] : C++上でros::init()を行うための引数

    ###  arm controll  ###
    ### arm1    =  1.0 ###
    ### arm2    =  1.0 ###
    ### arm3    = -1.0 ###
    ### arm4    = -1.0 ###
    ### gripper = -1.0 ###
    pro_ctr.moveArm( 1.0, 1.0, -1.0, -1.0)

if __name__ == '__main__':
    try:
        test()
    except rospy.ROSInterruptException: pass

File: sobit_pro_library/test/test_arm_controll.cpp
#include <sobit_pro_library/sobit_pro_joint_controller.h>

int main(int argc, char *argv[]) {
    ros::init(argc, argv, "sobit_pro_joiunt_controller_test");
    sobit::SobitProJointController sobit_pro_ctr;

    /*  arm controll  */
    /* arm1    =  1.0 */
    /* arm2    =  1.0 */
    /* arm3    = -1.0 */
    /* arm4    = -1.0 */
    sobit_pro_ctr.moveArm( 1.0, 1.0, -1.0, -1.0);

    return 0;
}
